src/plugins/dataflow/components/dataflow-program.tsx

```diff
@@ -538,21 +544,36 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
   };
 
   private recordCase = () => {
+    const { recordIndex } = this.props;
+    const { programDataRate } = this.props.tileModel; //grab the program Sampling Rate to write TimeQuantized
+    //attributes order  - Time_Quantized as first column | Time_Actual | + # of nodes
     const aCase: ICaseCreation = {};
+    const timeQuantizedKey = "Time_Quantized";
+    const timeActualKey = "Time_Actual";
+
+    const recordTimeQuantized = (recordIndex * programDataRate) / 1000; //in seconds
+    aCase[timeQuantizedKey] = recordTimeQuantized;
+    if (recordIndex === 0) {
+      this.setState({startTimeActual: Date.now()});
+    }
+    const recordTimeActual = (Date.now() - this.state.startTimeActual) / 1000; //in seconds
+    (recordTimeActual >= 0) && (aCase[timeActualKey] = recordTimeActual);
+
     //loop through attribute (nodes) and write each value
     this.programEditor.nodes.forEach((node, idx) => {
-      const key = this.props.tileModel.dataSet.attributes[idx].id;
+      const key = this.props.tileModel.dataSet.attributes[idx + 2].id; //starts at index 2 to skip over time attribute
       aCase[key] = node.data.nodeValue as string;
     });
     addCanonicalCasesToDataSet(this.props.tileModel.dataSet, [aCase]);
   };
 
   private playbackNodesWithCaseData = (dataSet: any, playBackIndex: number) => {
+
     const currentCase = dataSet.getCaseAtIndex(playBackIndex);
     if (currentCase){
       const {__id__} = currentCase; //this is the id of the case we are looking at for each frame
       this.programEditor.nodes.forEach((node, idx) => { //update each node in the frame
-        const attrId = dataSet.attributes[idx].id;
+        const attrId = dataSet.attributes[idx + 2].id; //add 2 to skip timeQuantized + timeActual
```

Comment:
Rather than duplicating this logic in two places, you could make a local function and then call it from both places:
```typescript
  function getAttributeIdForNode(nodeIndex: number) {
    // add two to skip time attributes
    return dataSet.attributes[idx + 2].id
  }
```
and then there's only one place to change it if we change the number of additional attributes.